api/nodecontrol/client.go

```diff
@@ -0,0 +1,160 @@
+package nodecontrol
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	nodegen "github.com/synadia-io/nex/api/nodecontrol/gen"
+	"github.com/synadia-io/nex/node/actors"
+)
+
+var (
+	DefaultRequestTimeout = 5 * time.Second
+)
+
+type ControlAPIClient struct {
+	nc *nats.Conn
+}
+
+func NewControlApiClient(nc *nats.Conn) (*ControlAPIClient, error) {
+	return &ControlAPIClient{
+		nc: nc,
+	}, nil
+}
+
+func (c *ControlAPIClient) Auction(tags []string) (*nodegen.AuctionResponseJson, error) {
+	req := nodegen.AuctionRequestJson{}
+	req_b, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	msg, err := c.nc.Request(actors.AuctionSubject(), req_b, DefaultRequestTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := new(nodegen.AuctionResponseJson)
+	err = json.Unmarshal(msg.Data, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (c *ControlAPIClient) Ping() (*nodegen.NodePingResponseJson, error) {
+	msg, err := c.nc.Request(actors.PingSubject(), nil, DefaultRequestTimeout)
+	if err != nil {
+		return nil, err
+	}
+	resp := new(nodegen.NodePingResponseJson)
+	err = json.Unmarshal(msg.Data, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (c *ControlAPIClient) FindAgent() (*nodegen.AgentPingResponseJson, error) {
+	msg, err := c.nc.Request(actors.AgentPingSubject(), nil, DefaultRequestTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := new(nodegen.AgentPingResponseJson)
+	err = json.Unmarshal(msg.Data, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (c *ControlAPIClient) DirectPing(nodeId string) (*nodegen.NodePingResponseJson, error) {
+	msg, err := c.nc.Request(actors.DirectPingSubject(nodeId), nil, DefaultRequestTimeout)
+	if err != nil {
+		return nil, err
+	}
+	resp := new(nodegen.NodePingResponseJson)
+	err = json.Unmarshal(msg.Data, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (c *ControlAPIClient) DeployWorkload(nodeId string) (*nodegen.StartWorkloadResponseJson, error) {
+	req := nodegen.StartWorkloadRequestJson{}
+	req_b, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	msg, err := c.nc.Request(actors.DeploySubject(nodeId), req_b, DefaultRequestTimeout)
```

Comment:
the subject you get back from this will have a `*` in place where the namespace needs to be. We can't make requests on wildcard subjects.  There's a difficulty in using the subscription subject functions (which include wildcards) as the request subject functions.

e.g. we should have a `DeploySubscribeSubject(nodeId)` and then `DeployRequestSubject(namespace, nodeId)`